runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -161,7 +161,7 @@ public int hashCode() {
 
         CorfuTable.IndexRegistry<?, ?> EMPTY = new CorfuTable.IndexRegistry<Object, Object>() {
             @Override
-            public Optional<CorfuTable.Index<Object, Object, ? extends Comparable<?>>> get(CorfuTable.IndexName name) {
+            public <I extends Comparable<?>> Optional<CorfuTable.Index<Object, Object, I>> get(CorfuTable.IndexName name) {
```

Comment:
Why do you need this. Are you trying to force the compiler to error out the index function mismatch ?

Reply:
It used to be a named generic like this before I changed it to wildcard. My change however requires client to specifically cast to a type (e.g. I) and suppressing warnings. However in the old form (current PR brings it back) provided type in the definition (e.g. I) is chosen as the return value from registry's get method and no additional casting or suppressing warning is required on the client side.